tests/rptest/tests/read_replica_e2e_test.py

```diff
@@ -72,6 +73,50 @@ def start_producer(self):
             message_validator=is_int_with_prefix)
         self.producer.start()
 
+    @cluster(num_nodes=6)
+    @matrix(partition_count=[10])
+    def test_produce_is_forbidden(self, partition_count):
+        # Create original topic
+        self.start_redpanda(3, si_settings=self.si_settings)
+        spec = TopicSpec(name=self.topic_name,
+                         partition_count=partition_count,
+                         replication_factor=3)
+
+        DefaultClient(self.redpanda).create_topic(spec)
+
+        # Make original topic upload data to S3
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(spec.name, 'redpanda.remote.write', 'true')
+
+        # Make sure topic manifest is uploaded to S3
+        def s3_has_topic_manifest():
+            objects = list(
+                self.redpanda._s3client.list_objects(self.s3_bucket_name))
+            total_uploaded = 0
+            for o in objects:
+                if o.Key.endswith(
+                        "/topic_manifest.json") and self.topic_name in o.Key:
+                    return True
+            return False
+
+        wait_until(
+            s3_has_topic_manifest,
+            timeout_sec=
+            30,  #should be uploaded since cloud_storage_segment_max_upload_interval_sec=5
+            backoff_sec=5,
+            err_msg=
+            f"Not all data is uploaded to S3 bucket, is S3 bucket: {list(self.redpanda._s3client.list_objects(self.s3_bucket_name))}"
+        )
```

Comment:
can you explain why this is needed? in particular it seems to be a condition that may not always be true in practice.

ideally the property of a topic being a read replica is something that we can apply entirely locally without involving any other services like s3, much like we can change other properties on topics and observe that property being true immediately.

Reply:
during creation of a read replica topic redpanda goes to S3 and downloads the topic manifest. Redpanda needs to know original's topic partition count and initial revision id before creating the topic. If there's not corresponding topic manifest in S3, create topic for read replica will fail

Reply:
That makes sense, thanks for the explanation.

When designing these sort of integration tests it's usually a good idea to start from a position that most directly mimics real-world scenarios. That wouldn't necessarily mean that objects have been uploaded into s3 before a second cluster is created / topic in the second cluster is created (folks do wild stuff, like not reading directions).

If I understand correctly, checking that the objects have been uploaded into s3 is a way the test to insert a barrier prior to creating the second cluster and topic? If so, it would be nice to remove this barrier and use retries instead.

Would the following pseudo-code for a reformulated test be expected to work in a real deployment scenario:

```
def test():
    # create the first cluster and topic
    self.start_redpanda(3, si_settings=self.si_settings)
    DefaultClient(self.redpanda).create_topic(spec)
    rpk.alter_topic_config(spec.name, 'redpanda.remote.write', 'true')

    # create the second cluster (no barrier like checking on manifests being uploaded)
    second_cluster = RedpandaService(self.test_context)
    second_cluster.start(start_si=False)
    rpk_second_cluster = RpkTool(self.second_cluster)

    def create_read_replica_topic():
        rpk_second_cluster = RpkTool(self.second_cluster)
        conf = {
            'redpanda.remote.readreplica': self.s3_bucket_name,
        }
        rpk_second_cluster.create_topic(self.topic_name, config=conf)

    # wait until the read replica topic creation succeeds
    wait_until(create_read_replica_topic, ...)

    # proceed with normal test of writing to the read replica topic
```

Reply:
Yeas, this pseudo code should work. Will rewrite the test and only log s3 content if topic fails to be created